src/commands/run.rs

```diff
@@ -471,19 +472,41 @@ impl RunCommand {
                 }
 
                 let component = module.unwrap_component();
-
-                let command = wasmtime_wasi::bindings::Command::instantiate_async(
-                    &mut *store,
-                    component,
-                    linker,
-                )
-                .await?;
-                let result = command
-                    .wasi_cli_run()
-                    .call_run(&mut *store)
+                let result = if let Ok(command) =
+                    wasmtime_wasi::p3::bindings::Command::instantiate_async(
+                        &mut *store,
+                        component,
+                        linker,
+                    )
                     .await
-                    .context("failed to invoke `run` function")
-                    .map_err(|e| self.handle_core_dump(&mut *store, e));
+                {
+                    let mut promises = wasmtime::component::PromisesUnordered::new();
```

Comment:
FYI, no need to use `PromisesUnordered` if you only need to await one `Promise` at a time; you can use `Promise::get` in that case.

Reply:
Awesome, thanks for letting me know, I'll switch!